src/rust/src/lib.rs

```diff
@@ -169,3 +175,1499 @@ extern "C" fn ccxr_close_handle(handle: RawHandle) {
         let _file = File::from_raw_handle(handle);
     }
 }
+
+#[cfg(test)]
+pub mod tests {
+    pub mod params {

```

Comment:
Is the `pub mod params` needed here?

Reply:
I read the unit tests and decided to move these tests in the same file they are being tested upon, so now they are in params.rs under test module.